src/CurlUtil.cc

```diff
@@ -291,6 +308,9 @@ bool HeaderParser::Parse(const std::string &headers)
     else if (header_name == "X-Pelican-Broker") {
         m_broker = header_value;
     }
+    else if (header_name == "X-Osdf-X509" && header_value == "true") {
```

Comment:
I guess we control the Director so we control the incoming headers, but I think we've been bitten elsewhere by being case sensitive when we parse headers. Maybe lowercase each `header_name` and match against the lowercase string?

Reply:
There's an earlier call to "CanonicalizeHeaders", I believe, that puts all headers into a standardized casing. That allows us to avoid case-insensitive comparisons (which tend to clutter up the C++ a bit).